R/logger.R

```diff
@@ -91,7 +91,7 @@ Logger <- setRefClass(
                 stop(paste("message composer(passed as composer_f) must be function",
                            " with signature function(msg, ...)"))
             }
-            msg_composer <<- composer_f
+            assign("msg_composer", composer_f, envir = .self)
```

Comment:
Are you sure this works?  I would have expected the environment to be assigned as parent.frame(n = 1), otherwise it should do the same thing as a single assignment inside a function

Reply:
Not going to comment on each one, please be sure you are doing the right assignment.

Reply:
- I ran regular unit tests and sample application loggers and all are working fine.
- I am not expert with S4 but it is similar to using `self` in R6 (in R6 functions cannot access class members without `self` or `private` keywords) but I will apply your suggestion as it seems better

Reply:
- It seems that `self` is working here the same as R6 and it means an instance of the class object and not "current" environment check [ref class](https://stat.ethz.ch/R-manual/R-devel/library/methods/html/refClass.html)
> Alternatively, a method may be an external method. This is signalled by .self being the first formal argument to the method. The body of the method then works like any ordinary function. The methods are called like other methods (without the .self argument, which is supplied internally and always refers to the object itself). Inside the method, fields and other methods are accessed in the form .self$x. External methods exist so that reference classes can inherit the package environment of superclasses in other packages; see the section on “External Methods”.

- However I updated the assign method to avoid specific environment assignment and search in parents environment for needed variable

Reply:
ok, thanks!